### PR TITLE
JDK-8282382: Report glibc malloc tunables in error reports

### PR DESCRIPTION
**Pull request by tstuefe**
On Linux/glibc platforms, the hs-err file should report the malloc tunables, or at least MALLOC_ARENA_MAX, since that one can have a significant effect on footprint or malloc contention for a VM.

This patch adds this printout as a one-liner into hs-err files and jcmd VM.info reports:

```
Process Memory:                                                                    
...
glibc malloc tunables: MALLOC_PERTURB_=40, MALLOC_ARENA_MAX=1
```

If the tunables are default, we print `glibc malloc tunables: (default)`.

Thanks, Thomas

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282382](https://bugs.openjdk.java.net/browse/JDK-8282382): Report glibc malloc tunables in error reports

### Reviewers
 * [Zhengyu Gu](https://openjdk.java.net/census#zgu) (@zhengyu123 - **Reviewer**) ⚠️ Review applies to 83d38ff0cb7b8b060b420269cfe7395f3c5ec897
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7620/head:pull/7620` \
`$ git checkout pull/7620`

Update a local copy of the PR: \
`$ git checkout pull/7620` \
`$ git pull https://git.openjdk.java.net/jdk pull/7620/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7620`

View PR using the GUI difftool: \
`$ git pr show -t 7620`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7620.diff">https://git.openjdk.java.net/jdk/pull/7620.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/os/linux/os_linux.cpp
Comment (zhengyu123) [inline src/hotspot/os/linux/os_linux.cpp:2169]: You can merge these two `#ifdef __GLIBC__` blocks.

**Review by zhengyu123 — APPROVED** [review]
Just a minor suggestion. Otherwise, looks good.

File: src/hotspot/os/linux/os_linux.cpp
Comment (tstuefe) [reply src/hotspot/os/linux/os_linux.cpp:2169]: Will do.

**Comment by tstuefe** [comment]
Thanks Zhengyu!

**Comment by tstuefe** [comment]
Friendly ping. This is quite simple and I need a second review.

File: src/hotspot/os/linux/os_linux.cpp
Comment (dholmes-ora) [inline src/hotspot/os/linux/os_linux.cpp:2120]: else st->cr()? Rather than adding the cr() externally.

**Review by dholmes-ora — APPROVED** [review]
Seems okay. One minor query.

Thanks,
David

File: src/hotspot/os/linux/os_linux.cpp
Comment (tstuefe) [reply src/hotspot/os/linux/os_linux.cpp:2120]: > else st->cr()? Rather than adding the cr() externally.

The idea is to have output that can, if needed, be combined into larger single lines. At least that's what I usually do with functions that output small snippets.

But I do not have strong emotions here and can change it if you prefer.

**Comment by tstuefe** [comment]
> Seems okay. One minor query.
> 
> Thanks, David

Thanks David!

File: src/hotspot/os/linux/os_linux.cpp
Comment (dholmes-ora) [reply src/hotspot/os/linux/os_linux.cpp:2120]: There are arguments both ways so no strong enough feeling to make a change.

**Comment by tstuefe** [comment]
Thanks @zhengyu123 and  @dholmes-ora.

/integrate

**Pull request closed without merging**
